sdks/python/apache_beam/runners/worker/data_plane.py

```diff
@@ -195,8 +252,14 @@ def add_to_inverse_output(data):
               instruction_id=instruction_id,
               transform_id=transform_id,
               data=data))
-    return ClosableOutputStream(
-        add_to_inverse_output, flush_callback=add_to_inverse_output)
+    if self._data_buffer_time_limit > 0:
```

Comment:
It'd be better if one didn't have to choose one or the other. Would it be possible to compose them with wrappers? Otherwise, I'd make them the same class with both capabilities (optionally configured).